src/loaders.rs

```diff
@@ -31,6 +32,45 @@ fn safe_join(directory: &Path, template_name: &str) -> Option<PathBuf> {
     }
 }
 
+#[cached(
+    size = 128,          // Cache size
+    result = true,       // Cache Result type
+    key = "String",      // Use owned String as key
+    convert = r##"{ dirname.to_string() }"## // Convert &str to String
+)]
+fn get_app_template_dirs(py: Python<'_>, dirname: &str) -> Result<Vec<String>, PyErr> {
+    // Import django.apps
+    let apps_module = PyModule::import(py, "django.apps")?;
+
+    // Get the 'apps' object
+    let apps = apps_module.getattr("apps")?;
+
+    // Call get_app_configs() on apps
+    let app_configs = apps.call_method0("get_app_configs")?;
+
+    // Convert to iterator and process each app_config
+    let app_configs_iter = app_configs.try_iter()?;
+
+    let mut template_dirs = Vec::new();
+    for app_config_result in app_configs_iter {
+        match app_config_result {
+            Ok(app_config) => {
+                let path = app_config.getattr("path")?;
+                if !path.is_none() {
+                    let path_str: String = path.extract()?;
+                    let template_path = PathBuf::from(path_str).join(dirname);
+                    if template_path.is_dir() {
+                        template_dirs.push(template_path.to_string_lossy().into_owned());
+                    }
+                }
+            }
+            Err(e) => return Err(e),
+        }
+    }
```

Comment:
We can inline some code here and use the `?` operator more:

```suggestion
    let mut template_dirs = Vec::new();
    for app_config_result in app_configs.try_iter()? {
        let path = app_config_result?.getattr("path")?;
        if path.is_some() {
            let path_str: String = path.extract()?;
            let template_path = PathBuf::from(path_str).join(dirname);
            if template_path.is_dir() {
                template_dirs.push(template_path.to_string_lossy().into_owned());
            }
        }
    }
```